src/ILCompiler.Compiler/src/Compiler/MetadataGeneration.cs

```diff
@@ -109,14 +113,7 @@ private void Graph_NewMarkedNode(DependencyNodeCore<NodeFactory> obj)
             {
                 _typesWithEETypesGenerated.Add(eetypeNode.Type);
                 AddGeneratedType(eetypeNode.Type);
-
-                // If this is an instantiated non-canonical generic type, add it to the generic instantiations hashtable
-                if (eetypeNode.Type.HasInstantiation && !eetypeNode.Type.IsGenericDefinition)
-                {
-                    if (!eetypeNode.Type.IsCanonicalSubtype(CanonicalFormKind.Any))
-                        GenericsHashtable.AddInstantiatedTypeEntry(_nodeFactory, eetypeNode.Type);
-                }
-
+                GenericsHashtable.AddInstantiatedTypeEntry(_nodeFactory, eetypeNode.Type);
```

Comment:
This doesn't look like an improvement - when I read the new code, it just makes me pause and wonder why we call `AddInstantiatedTypeEntry` for a type that is not instantiated. Same for `AddExactMethodInstantiationEntry` below.

Btw: is the existing `eetypeNode.Type.IsCanonicalSubtype` check trying to prevent `__UniversalCanon` from going in this table? (We should not reach this code path for any other canonical EETypes.)